pyat/at/tracking/particles.py

```diff
@@ -125,36 +133,51 @@ def sigma_matrix(ring=None, twiss_in=None, emitx=None,
         assert espread is not None, 'espread must be defined'
 
     if ring:
+        cavPassFlag = numpy.any(
+                        numpy.array(
+                          [i.PassMethod == 'CavityPass' for i in ring]))
+        radPassFlag = numpy.any(
+                        numpy.array(
+                          ['Rad' in i.PassMethod for i in ring]))
+
+        if cavPassFlag and not radPassFlag and not flag:
+            raise AtError('Cannot compute 6D sigma matrix without '
+                          'radiation damping and without emittances')
+
         ld0, bd, ld = ring.get_optics()
         if ring.radiation and not flag:
-            print('Generating correlated sigma matrix using ohmi envelope')
+            warnings.warn(AtWarning('Generating correlated sigma matrix using '
+                                    'ohmi envelope'))
+
             emit0, beamdata, emit = ohmi_envelope(ring, refpts=[0])
             sig_matrix = emit.r66[0]
         elif flag:
-            print('Generating pseudo-correlated matrix '
-                  'from initial optics conditions')
+            warnings.warn(AtWarning('Generating pseudo-correlated matrix '
+                      'from initial optics conditions'))
             if ring.radiation:
-                print('Ignoring provided blength and '
-                      'calculating it based on espread')
+                warnings.warn(AtWarning('Ignoring provided blength and '
```

Comment:
Add `if blength is not None`?

Reply:
This is already inside `if flag` which only is True if `blength` is defined. It seems a bit redundant (forcing it to be defined and then ignoring it later) but simplifies the code a lot.